Updating java docs (#505)
matter-labs/zksync-web-era-docs: docs/api/java/getting-started.md

1. Connect to the zkSync network.
2. Deposit assets from Ethereum into zkSync.
3. Check balances.
4. Transfer and withdraw funds (native and ERC20 tokens).
5. Deploy a smart contract.

> 6. Deploy a smart contract with create2.

Reviewer: ```suggestion
6. Deploy a smart contract with `create2`.
```

public class Main {
    public static void main(String ...args) {
        long chainId = 123L;// Chainid of the ZkSync Era network

        Credentials credentials = Credentials.create("0x<private_key>");

        EthSigner signer = new PrivateKeyEthSigner(credentials, chainId);
    }
}
```

## ZkSync Era wallet

> You can connect the zkSync Era wallet for easier operations.

Reviewer: This example is missing an import for `ZkSyncWallet`

Reviewer 2: Thank you

Reviewer 3: Hope you like this!!1

public class Main {
    public static void main(String ...args) {
        long chainId = 123L;// Chainid of the ZkSync network

        Credentials credentials = Credentials.create("0x<private_key>");

        EthSigner signer = new PrivateKeyEthSigner(credentials, chainId);
    }
}
```

> ## ZkSync wallet

Reviewer: ```suggestion
## ZkSync Era wallet
```

> You can connect zkSync wallet for easier operations.

Reviewer: ```suggestion
You can connect the zkSync Era wallet for easier operations.
```

        ZkSyncWallet wallet = new ZkSyncWallet(zksync, signer, Token.ETH);
    }
}
```

## Transactions

ZkSync2 supports Ethereum's `Legacy` and `EIP-1155` transactions, except for deploying contracts.

### EIP712

## Deploy contract

With zkSync Era, you can deploy a contract, using the `create` method, by transforming the contract into binary and deploying it to the zkSync Era network.

- [Storage](https://github.com/zksync-sdk/zksync2-python/blob/master/examples/solidity/storage/Storage.sol): Contract without constructor.
- [Incrementer](https://github.com/zksync-sdk/zksync2-python/blob/master/examples/solidity/incrementer/Incrementer.sol): Contract with constructor.
- [Demo](https://github.com/zksync-sdk/zksync2-python/blob/master/examples/solidity/demo/Demo.sol): Contract that has a dependency on
  [Foo](https://github.com/zksync-sdk/zksync2-python/blob/master/examples/solidity/demo/Foo.sol) contract.

> There is a [user guide](https://github.com/zksync-sdk/zksync2-python/blob/master/examples/README.md) on how to compile Solidity smart contracts using the `zksolc`

Reviewer: this was successful

Reviewer: (apparently. :))

compiler. `zksolc` compiler generates a `combined.json` file that contains the bytecode and ABI of a smart contract. Those files are used in the following examples.

### Deploy contract (create2) [EIP-1014](https://eips.ethereum.org/EIPS/eip-1014)

Code:

```java
import io.zksync.abi.TransactionEncoder;
import io.zksync.crypto.signer.EthSigner;
import io.zksync.methods.request.Eip712Meta;
import io.zksync.methods.request.Transaction;
import io.zksync.protocol.ZkSync;
import io.zksync.protocol.core.ZkBlockParameterName;
import io.zksync.transaction.fee.Fee;
import io.zksync.transaction.type.Transaction712;
import io.zksync.utils.ContractDeployer;
import org.web3j.abi.datatypes.Address;
import org.web3j.protocol.core.methods.response.TransactionReceipt;
import org.web3j.protocol.core.methods.response.EthSendTransaction;
import io.zksync.transaction.response.ZkSyncTransactionReceiptProcessor;

import org.web3j.utils.Numeric;

import java.math.BigInteger;
import java.security.SecureRandom;

public class Main {
  public static void main(String... args) {
    ZkSync zksync; // Initialize client

>     Credentials credentials = Credentials.create("PRIVATE_KEY"); // Initialize signer

Reviewer: no import

    BigInteger chainId = zksync.ethChainId().send().getChainId();

>     EthSigner signer = new PrivateKeyEthSigner(credentials, chainId);

Reviewer: no import for PrivateKeyEthSigner

>     ZkSyncTransactionReceiptProcessor processor = new ZkSyncTransactionReceiptProcessor(zksync, DEFAULT_POLLING_FREQUENCY, DEFAULT_POLLING_ATTEMPTS_PER_TX_HASH);

Reviewer: these constants are coming up unassigned

    String binary = "0x<bytecode_of_the_contract>";

    byte[] salt = SecureRandom.getSeed(32);

    // Here we can precompute contract address before its deploying
    String precomputedAddress = ContractDeployer.computeL2Create2Address(new Address(signer.getAddress()), Numeric.hexStringToByteArray(binary), new byte[]{}, salt).getValue();

    BigInteger nonce = zksync

>       .ethGetTransactionCount(credentials.getAddress(), DefaultBlockParameterName.PENDING).send()

Reviewer: getAddress (credentials) and `DefaultBlockParameterName` coming up red

      .getTransactionCount();

    Transaction estimate = Transaction.create2ContractTransaction(
      credentials.getAddress(),
      BigInteger.ZERO,
      BigInteger.ZERO,
      binary,
      "0x",
      salt
    );

>     ZkTransactionFeeProvider feeProvider = new DefaultTransactionFeeProvider(zksync, Token.ETH);

Reviewer: no imports

    Transaction712 transaction = new Transaction712(
      chainId.longValue(),
      nonce,

>       feeProvider.getGasLimit(estimate),

Reviewer: no import

      estimate.getTo(),
      estimate.getValueNumber(),
      estimate.getData(),
      BigInteger.valueOf(100000000L),
      feeProvider.getGasPrice(),
      credentials.getAddress(),
      estimate.getEip712Meta()
    );

    String signature = signer.getDomain().thenCompose(domain -> signer.signTypedData(domain, transaction)).join();
    byte[] message = TransactionEncoder.encode(transaction, TransactionEncoder.getSignatureData(signature));

    EthSendTransaction sent = zksync.ethSendRawTransaction(Numeric.toHexString(message)).send();

>     TransactionReceipt receipt = processor.waitForTransactionReceipt(sent.getResult());

Reviewer: no try catch  or throw o waitFor..

Reviewer: The above code also have a few unused imports

import io.zksync.methods.request.Eip712Meta;
import io.zksync.protocol.core.ZkBlockParameterName;
import io.zksync.transaction.fee.Fee;

        ZkSyncWallet wallet = new ZkSyncWallet(zksync, signer, Token.ETH);
    }
}
```

## Transactions

> ZkSync2 supports Ethereum's `Legacy` and `EIP-1155` transaction except deploying contract.

Reviewer: ```suggestion
ZkSync2 supports Ethereum's `Legacy` and `EIP-1155` transactions, except for deploying contracts.
```

### EIP712

## Deploy contract

> With zkSync, you can deploy a contract using the create method, by simply building the contract into a binary format and deploying it to the zkSync network.

Reviewer: ```suggestion
With zkSync Era, you can deploy a contract, using the `create` method, by transforming the contract into binary and deploying it to the zkSync Era network.
```

- [Storage](https://github.com/zksync-sdk/zksync2-python/blob/master/examples/solidity/storage/Storage.sol): Contract without constructor.
- [Incrementer](https://github.com/zksync-sdk/zksync2-python/blob/master/examples/solidity/incrementer/Incrementer.sol): Contract with constructor.
- [Demo](https://github.com/zksync-sdk/zksync2-python/blob/master/examples/solidity/demo/Demo.sol): Contract that has a dependency on
  [Foo](https://github.com/zksync-sdk/zksync2-python/blob/master/examples/solidity/demo/Foo.sol) contract.

There is a [user guide](https://github.com/zksync-sdk/zksync2-python/blob/master/examples/README.md) on how to compile Solidity smart contracts using `zksolc`
compiler. `zksolc` compiler generates a `combined.json` file that contains the bytecode and ABI of a smart contract. Those files are used in the following examples.

> ### Deploy contract (Create 2) [EIP-1014](https://eips.ethereum.org/EIPS/eip-1014)

Reviewer: ```suggestion
### Deploy contract (create2) [EIP-1014](https://eips.ethereum.org/EIPS/eip-1014)
```

        String sentTransactionHash = zksync.ethSendRawTransaction(Numeric.toHexString(message)).send().getTransactionHash();

        // You can check transaction status as the same way as in Web3
        TransactionReceipt receipt = zksync.ethGetTransactionReceipt(sentTransactionHash).send().getTransactionReceipt();
    }
}
```

> ### Deploy contract (Create)

Reviewer: ```suggestion
### Deploy contract (create)
```

        TransactionReceipt receipt = wallet.execute(contractAddress, contractFunction).send();
    }
}
```

> ### Execute contract via Web3j generic Contract

Reviewer: ```suggestion
### Execute contract via Web3j generic contract
```

        TransactionReceipt receipt = wallet.withdraw("0x<receiver_address>", amount, token).send();
    }
}
```

## Wallet

## Contract interaction using wrapper

> ## Fee

Reviewer: ```suggestion
### Fee
```
